src/widgets/forms/autocomplete/autocompleteBar.ts

```diff
@@ -71,7 +70,7 @@ export async function renderAutocompleteControl (dom:HTMLDocument,
       creationArea.removeChild(decoratedAutocomplete)
       decoratedAutocomplete = undefined
     } else {
-      displayAutocomplete()
+      await displayAutocomplete()
```

Comment:
**thought:** Souch a small word can make a big difference. Yet we still do not have any tests for this component, which worries me.

**suggestion**: Let's add automated tests for any components we change.

I have made some good experience with [testing-library](https://testing-library.com/docs/dom-testing-library/intro), since it helps a lot to test UI as a user experiences it. We need to do this to ensure no regressions come up and we already have a lot of Bugs in SolidOS which we need to get under control. Continuously improving and extending the test base is the only way to escape this situation in my experience.

Reply:
I thought that Typescript type checking would solve this, distinguishing between things which return `x` and `Promise<x>`

Reply:
async functions always return Promise<x> (implicitly or explicitly), but this does not ensure whether the promise is awaited correctly. Anyway my suggestion is of a general nature: Let's at least start writing tests for issues we encounter, so that we will not encounter them again in the future. If there was an issue regarding a missing await we should have a test that turns red if the await is missing.

Reply:
Yes.  Makes sense a a good way to work.    We can't be the only people thinking this ... maybe it will turn up in a new version of tsc or eslint.